experimental/dds/tree/src/id-compressor/IdCompressor.ts

```diff
@@ -384,13 +387,15 @@ export class IdCompressor {
 	public constructor(
 		public readonly localSessionId: SessionId,
 		public readonly reservedIdCount: number,
-		attributionId?: AttributionId
+		attributionId?: AttributionId,
+		logger?: ITelemetryLogger
```

Comment:
should be able to make this line something like `private readonly logger?: ItelemetryLogger` and avoid the field declaration and assignment